### PR TITLE
HPCC-19203 Utilize DAFILESRV file access mechanism

### PR DESCRIPTION
**Pull request by rpastrana**
- Acquires file access artifact as part of file stream process
- Changes dafileserv request to new secure format

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

### CODE REVIEW

**Comment by rpastrana** [comment]
@jpmcmu please review.

File: DataAccess/src/main/java/org/hpccsystems/spark/thor/PlainConnection.java [hunk 1]
```diff
@@ -253,13 +253,15 @@ private String makeInitialRequest() {
...
+        + this.recordDefinition.getJsonInputDef().length()
+        + this.recordDefinition.getJsonOutputDef().length());
     sb.append(" \"node\" : ");
     sb.append("{\n \"kind\" : \"");
     sb.append((this.dataPart.isIndex())? "indexread"  : "diskread");
-    sb.append("\",\n \"fileName\" : \"");
-    sb.append(this.dataPart.getFilename());
+    sb.append("\",\n \"metaInfo\" : \"");
+    sb.append(this.dataPart.getFileAccessBlob());
+    sb.append("\",\n \"filePart\" : \"");
```
Comment (jpmcmu) [inline DataAccess/src/main/java/org/hpccsystems/spark/thor/PlainConnection.java]: Wanted to confirm that fileName changed to filePart here on dafilersrv side

**Comment by jpmcmu** [comment]
@rpastrana Looks good. I did have a few comments / questions

File: DataAccess/src/main/java/org/hpccsystems/spark/thor/PlainConnection.java (hunk 1 above)
Comment (rpastrana) [reply DataAccess/src/main/java/org/hpccsystems/spark/thor/PlainConnection.java]: Good observation, yes  metainfo replaces filename. The blob contains the filename among other info. Filepart is the id if the filepart we're interested in.

**Comment by rpastrana** [comment]
@jpmcmu please review code review changes

**Comment by jpmcmu** [comment]
@rpastrana Looks good

**Comment by rpastrana** [comment]
@richardkchapman ready to merge

**Pull request merged**